app.go

```diff
@@ -278,8 +281,33 @@ func (a *App) RunCommand(args []string) error {
 // Run the application and parse the command line arguments.
 // This method blocks.
 func (a *App) Run() (err error) {
+	// Create the readline instance.
+	rl, err := readline.NewEx(&readline.Config{
+		Prompt:                 a.currentPrompt,
+		HistorySearchFold:      true, // enable case-insensitive history searching
+		DisableAutoSaveHistory: true,
+		HistoryFile:            a.config.HistoryFile,
+		HistoryLimit:           a.config.HistoryLimit,
+		AutoComplete:           newCompleter(&a.commands),
+		VimMode:                a.config.VimMode,
+	})
+	if err != nil {
+		return err
+	}
+	return a.RunWithReadline(rl)
+}
+
+func (a *App) RunWithReadline(rl *readline.Instance) (err error) {
 	defer a.Close()
 
+	rl.Config.Prompt = a.currentPrompt
+	rl.Config.HistorySearchFold = true
+	rl.Config.DisableAutoSaveHistory = true
+	rl.Config.HistoryFile = a.config.HistoryFile
+	rl.Config.HistoryLimit = a.config.HistoryLimit
+	rl.Config.AutoComplete = newCompleter(&a.commands)
+	rl.Config.VimMode = a.config.VimMode
```

Comment:
Instead of duplicating the config here, lets create a method App.readlineConfig() *readline.Config

Reply:
Yeah, I was thinkinking about it but I'm not sure if this can be accomplished - here I'm modyfing configuration of readline.Instance. Having App.readlineConfig wont help much as still I will have to rewrite those values one by one. But maybe you have an idea how to do it? ;)

Reply:
`a.rl.Config = a.readlineConfig()` Isn't that possible?

Reply:
Another option is to have a `App.setReadlineDefaults(*readline.Config)` that sets these values once

Reply:
Right :) I've just pushed changes